retrieval/target_provider.go

```diff
@@ -27,29 +28,37 @@ import (
 	clientmodel "github.com/prometheus/client_golang/model"
 
 	"github.com/prometheus/prometheus/config"
-	"github.com/prometheus/prometheus/utility"
 )
 
-const resolvConf = "/etc/resolv.conf"
+const (
+	sdTypeDNS  = "dns"
+	sdTypeEtcd = "etcd"
+
+	defaultResolvConf = "/etcd/resolv.conf"
+)
 
 var (
-	dnsSDLookupsCount = prometheus.NewCounter(
+	sdLookupsCount = prometheus.NewCounterVec(
 		prometheus.CounterOpts{
 			Namespace: namespace,
-			Name:      "dns_sd_lookups_total",
-			Help:      "The number of DNS-SD lookups.",
-		})
-	dnsSDLookupFailuresCount = prometheus.NewCounter(
+			Name:      "sd_lookups_total",
```

Comment:
I think we'd want each discovery methods to have it's own metrics, as makes sense for each.